fs/config/config.go

```diff
@@ -120,59 +120,132 @@ func init() {
+		configDir = filepath.Join(appDataDir, "rclone")
+		configFile = findFile(configDir, configFileName)
+	} else {
+		fs.Debugf(nil, "Environment variable APPDATA is not defined and cannot be used as configuration location")
+	}
+	return
+}
+
+// Look for existing rclone.conf in subdirectory rclone of XDG config home
+// (see XDG Base Directory specification: https://specifications.freedesktop.org/basedir-spec/latest/).
+// Doing this also on Windows, for backwards compatibility reasons, except
+// not using it for new config!
+// ($XDG_CONFIG_HOME\rclone\rclone.conf)
+func findXDGConfig() (configDir string, configFile string) {
+	if xdgConfigDir := os.Getenv("XDG_CONFIG_HOME"); xdgConfigDir != "" {
+		configDir = filepath.Join(xdgConfigDir, "rclone")
+		configFile = findFile(configDir, configFileName)
+	}
+	return
+}
+
+// Look for existing rclone.conf in subdirectory .config/rclone of user's home directory
+// (~/.config/rclone/rclone.conf)
+func findDotConfigConfig(home string) (configDir string, configFile string) {
+	if home != "" {
+		configDir = filepath.Join(home, ".config", "rclone")
+		configFile = findFile(configDir, configFileName)
+	}
+	return
+}
+
+// Look for existing .rclone.conf (legacy hidden filename) in root of user's home directory
+// (~/.rclone.conf)
+func findOldHomeConfig(home string) (configDir string, configFile string) {
+	if home != "" {
+		configDir = home
+		configFile = findFile(home, hiddenConfigFileName)
+	}
+	return
+}
+
+// Return the path to the configuration file
+func makeConfigPath() string {
+	// Look for existing rclone.conf in prioritized list of known locations
+	// Also decide which directory to use as default location for creating a new
+	// config file, after checking all known locations without finding an existing.
+	var (
+		configDir        string
+		configFile       string
+		defaultConfigDir string
+	)
+	// <rclone_exe_dir>/rclone.conf
+	if _, configFile = findLocalConfig(); configFile != "" {
+		return configFile
+	}
+	// Windows: $AppData/rclone/rclone.conf
+	if runtime.GOOS == "windows" {
+		if configDir, configFile = findAppDataConfig(); configFile != "" {
+			return configFile
 		}
+		// On Windows this is the preferred default location for new config when no existing is found
+		defaultConfigDir = configDir
+	}
+	// $XDG_CONFIG_HOME\rclone\rclone.conf
+	if configDir, configFile = findXDGConfig(); configFile != "" {
+		return configFile
+	}
+	if runtime.GOOS != "windows" {
+		// On Unix this is the preferred default location for new config when no existing is found
+		defaultConfigDir = configDir
+	}
+	// ~/.config/rclone/rclone.conf
+	homeDir, homeDirErr := findHomeDir()
+	if configDir, configFile = findDotConfigConfig(homeDir); configFile != "" {
+		return configFile
+	}
+	if defaultConfigDir == "" {
+		// When $AppData on Windows and $XDG_CONFIG_HOME on Unix does not exist,
+		// fallback to this as the default location when no existing file is found
+		defaultConfigDir = configDir
```

Comment:
What exactly is `configDir` set to here? I think it will have been set to `""` by the call to `findDotConfigConfig`?

So I don't think this is right, but I'm not sure what it should be! The comment needs to say what it should be. `homeDir` maybe?

Reply:
If homeDir is "" then configDir will be "", and then if defaultConfigDir=="" it will just be set to "" once more.
Otherwise it will be "~/.config/rclone" regardless if rclone.conf was found there or not.
So I think this actually works as intended.

Regardless: I'm trying a little adjustment: Keeping a separate variable primaryConfigDir with the APPDATA path on Windows and XDG path on Unix, and then keeping the ~/.config/rclone path in another variable fallbackConfigDir, and then waiting to choose between them until they are actually needed at the end.